### ISSUE TITLE
TCP_NODELAY is not supported

### ISSUE THREAD

**Issue by ValdikSS** [OpenVPN/ovpn-dco-win#90]
As of 1.2.1, TCP_NODELAY socket option (`--tcp-nodelay` or `--socket-flags TCP_NODELAY`) doesn't do anything, silently.
OpenVPN skips is in case of DCO-win, and there's no API to set it in the kernel driver.

https://github.com/OpenVPN/openvpn/blob/a7f80d402fb95df3c58a8fc5d12cdb8f39c37d3e/src/openvpn/socket.c#L957-L961
```
static bool
socket_set_flags(socket_descriptor_t sd, unsigned int sockflags)
{
    /* SF_TCP_NODELAY doesn't make sense for dco-win */
    if ((sockflags & SF_TCP_NODELAY) && (!(sockflags & SF_DCO_WIN)))
```

**Comment by lstipakov** [comment]
@cron2 Shall we just enable it by default?

**Comment by HqT65W6R** [comment]
Yes, please! Another option is to simply copy existing settings from hardware adapter used for OpenVPN connection.

**Comment by cron2** [comment]
Coming back to this, I wonder if this has any effect on where we sit in the kernel (on Linux kernel, it has no effect - generated packets by the DCO driver do not use the "classic" `send()` abstraction but something `skb_send_sock_locked_...()` deep in the stack).

... and whether we can just close it? ;-)

**Comment by ValdikSS** [comment]
@cron2 I'll test it and report back if it doesn't work as intended.
